### DISCUSSION TITLE
必须要在submit里面才能获取组件的值吗， 组件onChange时候有没有回调钩子什么的？

### DISCUSSION THREAD

**Discussion by fongfai** [alibaba/formily#1902] [Q&A 2.x] [answered]
比如这个示例(https://codesandbox.io/s/6jm7i?file=/App.tsx)里的 Submit，能不能实时获取输入框组件onChange的值

onFieldChange 里只能绑定一个field， 我的意思是有没有类似onFieldsChange， 就是像ant-form一样 拿到所有values

**Comment by janryWang** [comment, marked as answer]
麻烦好好看文档，form.values

**Reply by fongfai** [reply]
抱歉， 一开始没看到，着急用就先问下了，看到文档后想删除提问，发现disscution 无法删除~

**Reply by janryWang** [reply]
没事，以后尽量先找文档，提过的问题，实在找不到再提问

**Reply by ghost** [reply]
> form.values

我看 form.values 是个 proxy 对象,  怎么拿到普通对象
